src/services/files.js

```diff
@@ -174,6 +174,29 @@ const updateStatus = async (fileId, fileStatus) => {
   }
 };
 
+/**
+ * Updates the status of a file in the database based on its key.
+ * @param {string} key - The key of the file to update.
+ * @param {string} fileStatus - The new status of the file.
+ * @returns {Promise<Object|Error>} - A promise that resolves to the updated file object or
+ * an error object if an error occurs.
+ */
+const updateStatusByKey = async (key, fileStatus) => {
+  let file;
+  try {
+    // Update the file's status in the database using the File model
+    file = await File.update({ status: fileStatus }, {
+      where: { key },
+      individualHooks: true,
+    });
+    // Return the data values of the updated file
+    return file.dataValues;
```

Comment:
optional: reading the sequelize docs, I came to the conclusion that file.toJSON() actually might be a better solution generally than dataValues, since that way sequelize does the conversion and includes all the virtual fields. Doesn't matter here, but I think it might be a better habit to get into